### DISCUSSION TITLE
Need for jest-canvas-mock in jest tests for those using react-component-libarary

### DISCUSSION THREAD

**Discussion by s24y85u2** [Royal-Navy/design-system#1698] [Q&A] [answered]
When testing a component that renders a NumberInput component from the react-component-library, there is a need to include jest-canvas-mock into the devDependencies of the project due to the canvas context not being available in jest.

The issue arises in the following commit in version 2.16.0

https://github.com/Royal-Navy/design-system/commit/ae4f632761544f20933178053003cbc8cf1be416

At minimum the documentation or changelog should include references to this need for jest-canvas-mock when testing using jest.

### Steps to reproduce
1. Install react-component-library 2.16.0 or upwards
2. Create a component that renders a NumberInput
3. Create a jest test for that component
4. Run jest test and notice error

**Comment by m7kvqbe1** [comment, marked as answer]
Did you spawn an application using the CRA template package? This is included in the jest setup configuration:

https://github.com/Royal-Navy/design-system/commit/227a3f36858299062193d45034d93795ffc09873

**Reply by m7kvqbe1** [reply]
`setupTests.ts`
```typescript
import 'jest-canvas-mock'
```

**Comment by m7kvqbe1** [comment]
I'm closing this issue and converting it to a discussion so that it's available for future reference.
